pkg/generate/git/repository.go

```diff
@@ -337,6 +348,42 @@ func command(w io.Writer, name, dir string, env []string, args ...string) (stdou
 	return
 }
 
+// timedCommand executes an external command in the given directory with a timeout.
+// The command's standard out and error are trimmed and returned as strings
+// It may return the type *GitError if the command itself fails.
+func timedCommand(timeout time.Duration, w io.Writer, name, dir string, env []string, args ...string) (stdout, stderr string, err error) {
+	// typedWrapper wraps `command()` and returns a single, typed value in order for us to pipe it into a channel
+	typedWrapper := func(w io.Writer, name, dir string, env []string, args ...string) cmdResult {
+		stdout, stderr, err := command(w, name, dir, env, args...)
+		return cmdResult{
+			err:    err,
+			stdout: stdout,
+			stderr: stderr,
+		}
+	}
+
+	done := make(chan cmdResult)
+	go func() {
+		done <- typedWrapper(w, name, dir, env, args...)
+	}()
+
+	select {
+	case result := <-done:
+		// the command returned normally
+		return result.stdout, result.stderr, result.err
+	case <-time.After(timeout):
+		// the command timed out
+		return "", "", fmt.Errorf("execution of %s %s timed out after %s", name, strings.Join(args, " "), timeout)
```

Comment:
I'd prefer to have a known variable returned (ie. ErrTimeout) so that clients can compare and identify the error. The message is maybe good for logging.